localstack-core/localstack/services/events/archive.py

```diff
@@ -0,0 +1,167 @@
+
+    @property
+    def archive_arn(self) -> Arn:
+        return self.archive.arn
+
+    def set_state(self, state: ArchiveState) -> None:
+        self.archive.state = state
+
+    def set_creation_time(self) -> None:
+        self.archive.creation_time = datetime.now(timezone.utc)
+
+    def update(
+        self,
+        description: ArchiveDescription,
+        event_pattern: EventPattern,
+        retention_days: RetentionDays,
+    ) -> None:
+        self.set_state(ArchiveState.UPDATING)
+        if description is not None:
+            self.archive.description = description
+        if event_pattern is not None:
+            self.archive.event_pattern = event_pattern
+        if retention_days is not None:
+            self.archive.retention_days = retention_days
+        self.set_state(ArchiveState.ENABLED)
+
+    def delete(self) -> None:
+        # TODO handle active replays
+        self.set_state(ArchiveState.DISABLED)
+        try:
+            self.client.remove_targets(
+                Rule=self.rule_name, EventBusName=self.event_bus_name, Ids=[self.target_id]
+            )
+        except Exception as e:
+            LOG.debug(f"Target {self.target_id} could not be removed, {e}")
+        try:
+            self.client.delete_rule(Name=self.rule_name, EventBusName=self.event_bus_name)
+        except Exception as e:
+            LOG.debug(f"Rule {self.rule_name} could not be deleted, {e}")
+
+    def put_events(self, events: FormattedEventList) -> None:
+        for event in events:
+            self.archive.events[event["id"]] = event
+
+    def get_events(self, start_time: Timestamp, end_time: Timestamp) -> FormattedEventList:
+        events_to_replay = self._filter_events_start_end_time(start_time, end_time)
+        return events_to_replay
+
+    def _initialize_client(self) -> BaseClient:
+        client_factory = connect_to(aws_access_key_id=self.account_id, region_name=self.region)
+        client = client_factory.get_client("events")
+
+        service_principal = ServicePrincipal.events
+        client = client.request_metadata(service_principal=service_principal, source_arn=self.arn)
+        return client
+
+    def _create_archive_rule(
+        self,
+    ) -> RuleName:
+        rule_name = f"Events-Archive-{self.name}"
+        default_event_pattern = {
+            "replay-name": [{"exists": False}],
+        }
+        if self.event_pattern:
+            updated_event_pattern = json.loads(self.event_pattern)
+            updated_event_pattern.update(default_event_pattern)
+        else:
+            updated_event_pattern = default_event_pattern
+        self.client.put_rule(
+            Name=rule_name,
+            EventBusName=self.event_bus_name,
+            EventPattern=json.dumps(updated_event_pattern),
+        )
+        return rule_name
+
+    def _create_archive_target(
+        self,
+    ) -> TargetId:
+        target_id = f"Events-Archive-{self.name}"
+        self.client.put_targets(
```

Comment:
although the target is not used to send events to the event bus to remove load from the gateway a dedicated target is required to access parameters of the target in the provider during sending events to the archive

Reply:
documented in: https://github.com/localstack/localstack/pull/10890/commits/647b7a991b73485a64b005b775dbc0d43f40e7f8